Source/Details/ASDataController.h

```diff
@@ -160,17 +155,30 @@ extern NSString * const ASCollectionInvalidUpdateException;
  */
 @interface ASDataController : NSObject
 
-- (instancetype)initWithDataSource:(id<ASDataControllerSource>)dataSource eventLog:(nullable ASEventLog *)eventLog NS_DESIGNATED_INITIALIZER;
+- (instancetype)initWithDataSource:(id<ASDataControllerSource>)dataSource node:(nullable id<ASRangeManagedNode>)node eventLog:(nullable ASEventLog *)eventLog NS_DESIGNATED_INITIALIZER;
+
+- (instancetype)init NS_UNAVAILABLE;
+
+/**
+ * The node that owns this data controller, if any.
+ *
+ * NOTE: Soon we will drop support for using ASTableView/ASCollectionView without the node, so this will be non-null.
+ */
+@property (nonatomic, nullable, weak, readonly) id<ASRangeManagedNode> node;
 
 /**
  * The map that is currently displayed. The "UIKit index space."
+ *
+ * This property will only be changed on the main thread.
  */
-@property (nonatomic, strong, readonly) ASElementMap *visibleMap;
+@property (atomic, copy, readonly) ASElementMap *visibleMap;
```

Comment:
We don't need asserts to check this. We only change it internally. And we can't override atomic accessors so it's impossible to assert.